docs/JAX_transformer_text_classification.md

```diff
@@ -0,0 +1,355 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.15.2
+kernelspec:
+  display_name: jax-env
+  language: python
+  name: python3
+---
+
+# Text classification with transformer model
+
+```{code-cell} ipython3
+import io
+import typing
+
+import jax
+import jax.numpy as jnp
+from flax import nnx
+import optax
+import grain.python as grain
+import tqdm
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+```
+
+## Dataset
+
+```{code-cell} ipython3
+def prepare_imdb_dataset(num_words: int) -> tuple:
```

Comment:
A few bits here are taken from https://github.com/keras-team/keras/blob/495a570c5350e689d686c572009811cc4492d8c6/keras/src/datasets/imdb.py#L13 to fully reproduce data setup of [a Keras tutoral](https://keras.io/examples/nlp/text_classification_with_transformer/). I will simplify it.

Reply:
I revisited dataset preprocessing. Done!